src/dragonball/src/dbs_boot/src/x86_64/mptable.rs

```diff
@@ -276,6 +296,22 @@ pub fn setup_mptable<M: GuestMemory>(mem: &M, boot_cpus: u8, max_cpus: u8) -> Re
         mpc_intsrc.0.srcbusirq = i;
         mpc_intsrc.0.dstapic = ioapicid;
         mpc_intsrc.0.dstirq = i;
+        // Patch irq routing entry for mptable if it is registered
+        // as PCI legacy irq.
+        if let Some(irq_device) = pci_legacy_irqs {
+            if irq_device.contains_key(&i) {
+                let device_id = irq_device.get(&i).unwrap();
```

Comment:
```suggestion
            if let Some(device_id) = irq_device.get(&i) {
```

These two lines are able to merged into one.

Reply:
done, nice catch